test/idp/src/main/kotlin/org/codice/compliance/web/sso/RedirectLoginTest.kt

```diff
@@ -34,27 +33,32 @@ import org.codice.compliance.verification.profile.SingleSignOnProfileVerifier
 import org.codice.security.saml.SamlProtocol
 import org.codice.security.sign.Encoder
 import org.codice.security.sign.SimpleSign
-import java.time.Instant
+import org.joda.time.DateTime
+import org.opensaml.saml.common.SAMLVersion
+import org.opensaml.saml.saml2.core.impl.AuthnRequestBuilder
+import org.opensaml.saml.saml2.core.impl.IssuerBuilder
 
 class RedirectLoginTest : StringSpec() {
-    companion object {
-        fun setupAuthnRequest(relayState: String?): Map<String, String> {
-            val baseRequest = getResource("redirect-authn-request.xml").readText()
-            val encodedRequest = Encoder.encodeRedirectMessage(String.format(baseRequest,
-                    ACS_URL,
-                    Common.getSingleSignOnLocation(SamlProtocol.REDIRECT_BINDING),
-                    ID,
-                    Instant.now().toString(),
-                    SP_ISSUER))
-            return SimpleSign().signUriString(SAML_REQUEST, encodedRequest, relayState)
-        }
-    }
-
     init {
         RestAssured.useRelaxedHTTPSValidation()
 
         "Redirect AuthnRequest Test" {
-            val queryParams = setupAuthnRequest(null)
+            val authnRequest = AuthnRequestBuilder().buildObject().apply {
```

Comment:
✏️ Just like in the post tests, these blocks to build the `authnRequest` look suspiciously similar.